### DISCUSSION TITLE
How to do configuration for auth ?

### DISCUSSION THREAD

**Discussion by luddwskater** [SAP/luigi#2884] [Q&A]
I tried to fetch client_id and authority from json file. This doesn't work as  appsettingsProvider.getClient() calls from Luigi before fetch for json file is ready. Is any way to get value for client_id ... from json file ?
import { appsettingsProvider } from './appsettings-provider';

class Auth {
 
    use = 'oidcAuth';
    
    storage = 'localStorage'; // localStorage, sessionStorage, none

  disableAutoLogin = false;

  oidcAuth = {
 
      idpProvider: OpenIdConnect,
      authority:  appsettingsProvider.getPinguri(),
   RequirePkce: true,
  response_mode: "fragment",
  loadUserInfo: true,
        client_id: appsettingsProvider.getClient(),
  redirect_uri: appsettingsProvider.getRedirectUri(),

**Comment by JohannesDoberer** [comment]
Hi @luddwskater , I don't know what appsettingsProvider does, hence I can only give a general advice.
If your luigi config needs values from async operations like a fetch call, just collect them before calling Luigi.setConfig(...), either promise style or with async/await.
E.g.
```
fetch('path/to/some.json')
  .then((response) => response.json())
  .then((data) => {
     Luigi.setConfig({
        someConfigProp: data.someConfigValue
     });
   });
   ```
